pkgs/development/libraries/agda/agda-categories/depend.patch

```diff
@@ -0,0 +1,9 @@
+diff --git a/agda-categories.agda-lib b/agda-categories.agda-lib
+index cc76242..888b196 100644
+--- a/agda-categories.agda-lib
++++ b/agda-categories.agda-lib
+@@ -1,3 +1,2 @@
+ name: agda-categories
+-depend: standard-library
```

Comment:
Perhaps `AgdaStdLib` is badly named so it can't detect it. Probably it'd not be too hard to figure out what this agda-lib stuff is looking for and make it so that AgdaStdLib dependency is picked up by it without changing this file.

Reply:
The problem is that the default for all the agda-lib stuff is `~/.agda` which isn't a very convenient location for nix. I believe this can be modified by the `AGDA_DIR` environment variable so I might have a play around at some point if I have time. I wonder if wrapping agda to have this set to some nix generated folder would allow the agda builder to be simpler/ make an `agdaWithPackages` derivation very easy